ouroboros-consensus/src/Ouroboros/Consensus/Node/Run/Byron.hs

```diff
@@ -40,49 +40,64 @@ instance RunNode ByronBlock where
 
   -- The epoch size is fixed and can be derived from @k@ by the ledger
   -- ('kEpochSlots').
-  nodeEpochSize           = \_proxy cfg _epochNo -> return
-                              . (coerce :: EpochSlots -> EpochSize)
-                              . kEpochSlots
-                              . Genesis.gdK
-                              . extractGenesisData
-                              $ cfg
+  nodeEpochSize             = \_proxy cfg _epochNo -> return
+                                . (coerce :: EpochSlots -> EpochSize)
+                                . kEpochSlots
+                                . Genesis.gdK
+                                . extractGenesisData
+                                $ cfg
+
+  nodeMaxBlockSize          = Aux.getMaxBlockSize . byronLedgerState
+  nodeBlockEncodingOverhead =
+    let safetyMargin        = 1024 {- Just for safety -}
+        blockHeaderOverhead = 650 {- the maximum block header size -}
+        blockBodyOverhead   = 1 {- ABody: encodeListLen 4 -}
+                            + 2 {- TxPayload: list -}
+                            + 1 {- SscPayload: encodeListLen 2 -}
+                            + 1 {- SscPayload: Word8 -}
+                            + 1 {- SscPayload: mempty :: Set () -}
+                            + 2 {- Delegation.Payload: list -}
+                            + 1 {- Update.Payload: encodeListLen 2 -}
+                            + 1 {- Update.Payload: Maybe AProposal -}
+                            + 2 {- Update.Payload: list of AVote -}
+    in const (blockHeaderOverhead + blockBodyOverhead + safetyMargin)
```

Comment:
Best to move this calculation to a separate function in `Ouroboros.Consensus.Ledger.Byron.Forge`, we will probably need it too for the hard fork combinator.